Tests/Weaver/Realm.Fody.Tests/AnalyticsTests.cs

```diff
@@ -104,19 +104,45 @@ public void Submit_WhenDisabled_PayloadIsEmpty()
             }
         }
 
-        private void ValidateAnalyticsPayload<T>(string featureName, T expectedResult)
+        [Test]
+        public void ValidateHostOS()
         {
-            foreach (var framework in _frameworks.Value)
+            var currentOs = Environment.OSVersion;
+            try
             {
-                var response = WeaveRealm(framework, "DryRun");
-                var payload = BsonSerializer.Deserialize<BsonDocument>(response).AsBsonDocument;
-                Assert.That(payload[featureName].AsString, Is.EqualTo(expectedResult.ToString()), $"Feature {featureName} was not reported as used: {expectedResult} in {framework}");
+                CompileAnalyticsProject();
+                ValidateAnalyticsPayloadAllFrameworks(new[] {
+                    (featureName: Metric.Environment.HostOsType, expectedValue: ConvertToMetricOS(currentOs.Platform)),
+                    (featureName: Metric.Environment.HostOsVersion, expectedValue: currentOs.Version.ToString()) });
+            }
+            catch (Exception e)
+            {
+                Assert.Fail($"{e.Message}");
             }
         }
 
-        private void ValidateSdkApiAnalyticsPayload(string featureName, byte expectedUsed = 1)
+        private static string ConvertToMetricOS(PlatformID platformID) =>
+            platformID switch
+            {
+                PlatformID.Win32NT or PlatformID.Win32S or PlatformID.Win32Windows or PlatformID.WinCE => Metric.OperatingSystem.Windows,
+                PlatformID.MacOSX => Metric.OperatingSystem.MacOS,
+                PlatformID.Unix => Metric.OperatingSystem.Linux,
+                _ => platformID.ToString()
+            };
+
+        private void ValidateAnalyticsPayloadAllFrameworks<T>((string featureName, T expectedValue)[] payloadFeatures)
```

Comment:
Rather than have this be generic, can we make `expectedValue` a string and have the caller pass in the correct value?